File: desh/ege2026kp/12solve/12-319.cpp
// Автор: М. Нехорошева

#include<iostream>
#include<string>
using namespace std;
int main(){
	int n=6;
	string s=(string(15,'3')+string(18,'2')+string(n,'1'));
//	cout<<s<<endl;
	for(;s.find("31")!=-1  || s.find("33")!=-1 || s.find("21")!=-1;){
		if(s.find("31")!=-1)
			s.replace(s.find("31"),2,"123");
		if(s.find("33")!=-1)
			s.replace(s.find("33"),2,"211");
		if(s.find("21")!=-1)
			s.replace(s.find("21"),2,"1");
	}
	int sum=0;
	for(int i=0;i<s.size();i++)
		sum+=s[i]-'0';
	if(sum>24)
		cout<<n;
	else cout<<"NO";
}

File: desh/ege2026kp/12solve/12-319.py
n = 1
while True:
  s = 15*'3' + 18*'2' + n*'1'
  while '31' in s or '33' in s or '21' in s:
    s = s.replace( '31', '123', 1)
    s = s.replace( '33', '211', 1)
    s = s.replace( '21', '1', 1)
  sumDigits = sum( map(int, s) )
  if sumDigits > 24:
    print( n )
    break
  n += 1

# Автор: Е. Джобс

for n in range(1,1000):
  s = '3'*15 + '2'*18 + '1'*n
  while '31' in s or '33' in s or '21' in s:
    s = s.replace( '31', '123', 1)
    s = s.replace( '33', '211', 1)
    s = s.replace( '21', '1', 1)
  if sum(map(int, s)) > 24:
    print( n )
    break
